test/opcode/varargs.lua

```diff
@@ -0,0 +1,55 @@
+--[[----------------------------------------------------------------------------
+ 
+  (c) Copyright IBM Corp. 2017
+ 
+   This program and the accompanying materials are made available
+   under the terms of the Eclipse Public License v1.0 and
+   Apache License v2.0 which accompanies this distribution.
+ 
+       The Eclipse Public License is available at
+       http://www.eclipse.org/legal/epl-v10.html
+ 
+       The Apache License v2.0 is available at
+       http://www.opensource.org/licenses/apache2.0.php
+ 
+  Contributors:
+     Multiple authors (IBM Corp.) - initial implementation and documentation
+ 
+--]]----------------------------------------------------------------------------
+
+--[[
+  Tests for varargs opcode.
+  
+  Other opcodes exercised by this function:
+    - CALL, GETTABUP, LOADK, MOVE, RETURN
+--]]
+
+-- setup
+local asserts = require "asserts"
+local jit = require "lvjit"
+local assert_equal = asserts.assert_equal
+local assert_compile = asserts.assert_compile
+jit.setjitflags(assert_equal, jit.JITBLACKLIST)
+jit.setjitflags(assert_compile, jit.JITBLACKLIST)
+
+local function varargs1(...)
+   local a = select(1,...)
+   local b = select(2,...)
+   local c = select(1,...)
+   local d = select(3,...)
+   local e = select(2,...)
+   local f = select(4,...)
+   local g = select("#", ...)
+   return a,b,c,d,e,f,g
+end
+
+assert_compile(varargs1, "varargs1")
+jit.setjitflags(varargs1, jit.BLACKLIST)
+local a,b,c,d,e,f,g = varargs1(7,4,9)
+assert_equal(7, a, "first value of varargs1()")
+assert_equal(4, b, "second value of varargs1()")
+assert_equal(7, c, "third value of varargs1()")
+assert_equal(9, d, "fourth value of varargs1()")
+assert_equal(4, e, "fifth value of varargs1()")
+assert_equal(nil, f, "sixth value of varargs1()")
+assert_equal(3, g, "seventh value of varargs1()")

```

Comment:
Could you use slightly more descriptive names rather than just letters? Just so it's easier to keep track of what value should be in each.

I'd be happy with just `arg1`, `arg2`, `arg1_2`, `size`, etc.